base/base32768_body.hpp

```diff
@@ -229,23 +229,22 @@ void Base32768Encode(Array<std::uint8_t const> input,
     std::int32_t code_point = (data & mask) >> shift;
     CHECK_LE(0, code_point);
     CHECK_LT(code_point, 1 << bits_per_code_point);
-    std::memcpy(
-        output.data, &(repertoire->Encode(code_point)), sizeof(char16_t));
+    *output.data = repertoire->Encode(code_point);
 
     // The following computation may cause |input.data| to overshoot the end if
     // using the special encoding at the end.  This is safe as soon as the loop
     // condition uses <.
     input_bit_index += bits_per_code_point;
     input.data += input_bit_index / bits_per_byte;
     input_bit_index %= bits_per_byte;
-    output.data += sizeof(char16_t);
+    ++output.data;
   }
 }
-UniqueArray<std::uint8_t> Base32768Encode(Array<std::uint8_t const> input,
-                                          bool const null_terminated) {
+UniqueArray<char16_t> Base32768Encode(Array<std::uint8_t const> input,
```

Comment:
`input` can be `const` here.

Reply:
`input.data += input_bit_index / bits_per_byte`.